config/routes.rb

```diff
@@ -202,6 +201,8 @@
         get '/jurisdiction' => 'jurisdiction#new'
         post '/jurisdiction' => 'jurisdiction#create'
         get '/jurisdiction/failure/:reason' => 'jurisdiction#failure', as: :jurisdiction_failure
+        get '/cancel/:step' => 'cancellations#new', as: :cancel_step
```

Comment:
This is one of the things I'm unsure about. We are putting the step in the path so we can know which step to route the user back to. The other option is to put the go back step in the session. I'm 50/50 on which is better so hoping to get some thoughts from reviewers.

Reply:
I'm always uncomfortable putting information in the user session that is tied to the page they're on. In general, it's bad practice because it prevents people from having multiple tabs open on the same site with the same login session (e.g., consider how something like Google Docs might behave if its state was stored in the user session and not scoped to the document). It can also break the back button.

One way I've done this before is to store state in a secure fashion in the page as a hidden variable. This ties it to where they are in the process (the page) so the back button is not broken. It lets them do multiple things at once on the site (unlikely for login.gov). It makes the state information tamper-proof (encryption). Not relying on the URL to tell us their state prevents someone from jumping into the middle of a process if we want to control the flow through that process.

Generally, the user session should hold data that applies to all pages they access regardless of what they are trying to do (it's global data). Contextual or local data should be managed outside of the global data store/user session.